### PR TITLE
Feature/task 170 catia import JSON

### PR DESCRIPTION
**Pull request by franzTobiasDLR**
Adds import functionality from a CATIA JSON file

### CODE REVIEW

File: de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporter.java [hunk 1]
```diff
@@ -9,33 +9,85 @@
...
+	public Command transform(EditingDomain editingDomain, JsonObject jsonObject,
+			Map<String, StructuralElementInstance> mapJsonUuidToSEI) {
+
+		CompoundCommand importCommand = new CompoundCommand();
+		this.editingDomain = editingDomain;
+
+		boolean worked = true;
+
+		// Import parts
+		for (JsonObject part : CatiaHelper.getListOfAllJSONParts(jsonObject)) {
```
Comment (andriikovalov-dlr) [inline de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporter.java]: If `mapJsonUuidToSEI` is not used for filtering SEIs to update, than what's the point of passing it from outside? We could also create it directly in this method

File: de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java [hunk 2]
```diff
@@ -37,19 +50,20 @@
...
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+import de.dlr.sc.virsat.project.structure.VirSatProjectCommons;
+import de.dlr.sc.virsat.project.test.AProjectTestCase;
 
 /**
  * The CATIA importer test class
  *
  */
-public class CatiaImporterTest extends AConceptTestCase {
+public class CatiaImporterTest extends AProjectTestCase {
```
Comment (andriikovalov-dlr) [inline de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java]: Why don't we inherit AConceptProjectTestCase?

File: de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java [hunk 3]
```diff
@@ -67,28 +81,203 @@
...
+
+		JsonObject rootObject = createMappedJsonObjectWithProductAndConfiguration();
+
+		// Do the import
+		CatiaImporter importer = new CatiaImporter();
+		Map<String, StructuralElementInstance> mapping = importer.mapJsonUuidToSEI(rootObject, configurationTree);
+		Command importCommand = importer.transform(editingDomain, rootObject, mapping);
+		editingDomain.getVirSatCommandStack().execute(importCommand);
+
+		assertTrue("Check if product values are imported", elementConfigurationReactionWheel1
```
Comment (andriikovalov-dlr) [inline de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java]: let's maybe create a variable for `elementConfigurationReactionWheel1.getFirst(Visualisation.class)` ?

File: de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporter.java (hunk 1 above)
Comment (franzTobiasDLR) [reply de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporter.java]: Well, it is intended to be used outside. I just wrote a test case for that. Here I just missed filtering with the map....

File: de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java [hunk 4]
```diff
@@ -67,30 +78,340 @@
...
+				repository.getActiveConcepts().add(conceptPS);
+				repository.getActiveConcepts().add(conceptVis);
+			}
+		});
 		createTestTreeScenario();
 	}
 
+	@After
+	public void tearDown() throws CoreException {
+		super.tearDown();
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.mechanical.test/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporterTest.java]: Do we need the tear down?

**Pull request merged**